sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java

```diff
@@ -180,7 +230,80 @@ public void readEndToEnd() throws Exception {
         assertNotNull(testCosmosDocument.getEtag());
         assertNotNull(testCosmosDocument.getTimestamp());
 
-        // TODO: configure ProGuard to avoid deserialization errors and verify the content of the parsed document
+        TestDocument testDocument = testCosmosDocument.getDocument();
+        assertNotNull(testDocument);
+        assertEquals(TEST_FIELD_VALUE, testDocument.test);
+    }
+
+    @Test
+    public void readFailTokenExchangeReturnsFailedTokenResultPayload() throws Exception {
+        HttpClientRetryer httpClient = mock(HttpClientRetryer.class);
+        whenNew(HttpClientRetryer.class).withAnyArguments().thenReturn(httpClient);
+        when(SharedPreferencesManager.getBoolean(STORAGE_ENABLED_KEY, true)).thenReturn(false);
+        Storage storage = Storage.getInstance();
+        Channel channel = start(storage);
+
```

Comment:
extra line in multiple places... just just one more point out  :D

Reply:
I'll fix in the product code, but is it ok to keep it as in tests? @guperrot, what do you think?

Reply:
I think we should fix this when once we saw it... :)

Reply:
add comments what you mock or use the "if" "when" "then" test generic comments